server/src/main/java/org/elasticsearch/cluster/routing/RoutingTable.java

```diff
@@ -470,7 +464,7 @@ public Builder updateNumberOfReplicas(final int numberOfReplicas, final String[]
                 IndexRoutingTable.Builder builder = new IndexRoutingTable.Builder(indexRoutingTable.getIndex());
                 // re-add all the shards
                 for (int i = 0; i < indexRoutingTable.size(); i++) {
-                    builder.addIndexShard(indexRoutingTable.shard(i));
+                    builder.addIndexShard(new IndexShardRoutingTable.Builder(indexRoutingTable.shard(i)));
```

Comment:
Should we do `builder.ensureShardArray(indexRoutingTable.size()-1)` before this loop?

Reply:
Yea that sounds like it should save an allocation here and there :) Added it